lib/install.js

```diff
@@ -403,7 +403,9 @@ function install (gyp, argv, callback) {
    * the compilation will succeed...
    */
 
-  function eaccesFallback () {
+  function eaccesFallback (err) {
+    var noretry = '--node_gyp_internal_noretry'
+    if (-1 !== argv.indexOf(noretry)) return cb(err)
```

Comment:
_Non-blocking Bikeshed:_

Using a yoda expression and a one-line if statement seems odd here. The yoda is presumably to make it easier to read, but squashing onto one line makes it harder to read.

```js
if (-1 !== argv.indexOf(noretry)) return cb(err)
```

```js
if (argv.indexOf(noretry) !== -1) {
  return cb(err)
}
```

nbd, more curious than anything else.

Reply:
I like my guard statements on one line, mainly for greppability.